File: backend/app/models/schema_defs/watchlist.py
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class WatchlistCreate(BaseModel):
    symbol: str
    name: str = ""
    base_position: int = 1000
    available_position: int = 1000
    cost_basis: Optional[float] = None
    memo: str = ""


class WatchlistItemOut(BaseModel):
    symbol: str
    name: str
    base_position: int
    available_position: int
    cost_basis: Optional[float] = None
    memo: str
    created_at: datetime

File: frontend/src/types/watchlist.ts
import type { QuoteSnapshot, TradingRule } from "./market";
import type { StrategySuggestion } from "./analysis";

export interface WatchlistItem {
  symbol: string;
  name: string;
  base_position: number;
  available_position: number;
  cost_basis?: number | null;
  memo: string;
  created_at?: string;
}

export interface WatchlistSignal {
  symbol: string;
  name: string;
  base_position: number;
  available_position: number;
  cost_basis?: number | null;
  memo: string;
  signal: StrategySuggestion;
  quote: QuoteSnapshot;
  rules: TradingRule;
  error?: string | null;
}
